scripts/build_avado.sh

```diff
@@ -28,46 +28,39 @@ if [ -z "${1:-}" ]; then
   exit 1
 fi
 
+if [[ "${1}" == "-h" ]] || [[ "${1}" == "--help" ]]; then
+  usage
+  exit 0
+fi
+
+# Get/set default values of parameters
+declare avado_version="${1}"
 declare environment_id="${2:-"$(${mydir}/get-default-environment.sh)"}"
 declare release_id="${3:-"$(${mydir}/get-default-environment.sh --release)"}"
+declare api_token="${4:-"!5qxc9Lp1BE7IFQ-nrtttU"}" # <- Default AVADO API token
 
-if [ -z "${environment_id}" ]; then
-  msg "Could not determine default environment"
+# Validate environment and release ids
+if [[ -z "${environment_id}" ]] || [[ -z "${release_id}" ]]; then
+  msg "Could not determine default environment or release id"
+  exit 1
+fi
+
+# AVADO build tool requires proper semver versions
+if ! [[ $avado_version =~ [0-9]{1,}\.[0-9]{1,}\.[0-9]{1,}$ ]]; then
+  msg "Version '${avado_version}' is not semver"
   exit 1
 fi
 
 # Retrieve the provider URL for the given environment ID from protocol-config.json
-provider_url=$(jq -r ".environments.\"${environment_id}\".network_id as \$envid | .networks[\$envid].default_provider // \"\"" "${mydir}/../packages/core/protocol-config.json")
+declare provider_url=$(jq -r ".environments.\"${environment_id}\".network_id as \$envid | .networks[\$envid].default_provider // \"\"" "${mydir}/../packages/core/protocol-config.json")
 if [ -z "${provider_url}" ]; then
   msg "Environment ${environment_id} has invalid network_id"
   exit 1
 fi
 
-if [[ "${1}" == "-h" ]] || [[ "${1}" == "--help" ]]; then
-  usage
-  exit 0
-fi
-
-msg "Building Avado for release ${release_id} in environment ${environment_id} with default provider ${provider_url}"
-
-declare AVADO_VERSION="${1}"
-
-if ! [[ $AVADO_VERSION =~ [0-9]{1,}\.[0-9]{1,}\.[0-9]{1,}$ ]]; then
-  # AVADO build tool requires proper semver versions
-  msg "Version '${AVADO_VERSION}' is not semver"
-  exit 1
-fi
 
 cd "${mydir}/../packages/avado"
 
-declare default_development_release_id="master-goerli"
-
-if [[ -z $(grep -E "${default_development_release_id}" "./build/Dockerfile") ]]; then
-  # Fail if default environment is no longer `master-goerli`
-  msg "Avado Dockerfile differs. Could not find \"${default_development_release_id}\" release id in it"
```

Comment:
The idea was to fail here if someone changes the files such that the regex does no longer work but maybe not longer necessary

Reply:
The `master-goerli` was removed from the `Dockerfile` and it now uses `UPSTREAM_VERSION` build arg, thats supplied through the docker-compose.yml.